spec/controllers/catalog_controller_spec.rb

```diff
@@ -0,0 +1,219 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CatalogController, type: :controller do
+
+  # /get_next
+  describe "GET next_callnumber as xhr request and required params" do
```

Comment:
There are a few rspec style guides out there that might help with the organization of these specs. In particular: https://www.betterspecs.org/#contexts and https://github.com/rubocop/rspec-style-guide#use-contexts have recommendations that I try to follow. Contexts can also be nested. I usually go with something like:

```
describe "GET next_callnumber" do
    context "when request is valid" do
        it "returns a 200"
    end

   context "when required parameters are missing" do
       it "returns a 500"
   end
end

describe "GET previous_callnumber"
     context "1"

     context "2"

     context "...."
end

describe "GET build_carousel" ...
```